File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/AlertDialogItemsBuilder.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils;

import android.content.Context;
import android.content.DialogInterface.OnClickListener;

import com.google.android.material.dialog.MaterialAlertDialogBuilder;

import java.util.ArrayList;

public class AlertDialogItemsBuilder
{
  private Context mContext;

  private ArrayList<CharSequence> mLabels = new ArrayList<>();
  private ArrayList<OnClickListener> mListeners = new ArrayList<>();

  public AlertDialogItemsBuilder(Context context)
  {
    mContext = context;
  }

  public void add(int stringId, OnClickListener listener)
  {
    mLabels.add(mContext.getResources().getString(stringId));
    mListeners.add(listener);
  }

  public void add(CharSequence label, OnClickListener listener)
  {
    mLabels.add(label);
    mListeners.add(listener);
  }

  public void applyToBuilder(MaterialAlertDialogBuilder builder)
  {
    CharSequence[] labels = new CharSequence[mLabels.size()];
    labels = mLabels.toArray(labels);
    builder.setItems(labels, (dialog, i) -> mListeners.get(i).onClick(dialog, i));
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/AlertDialogItemsBuilder.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.utils

import android.content.Context
import android.content.DialogInterface.OnClickListener
import com.google.android.material.dialog.MaterialAlertDialogBuilder

class AlertDialogItemsBuilder(private val context: Context) {
    private val labels = ArrayList<CharSequence>()
    private val listeners = ArrayList<OnClickListener>()

    fun add(stringId: Int, listener: OnClickListener) {
        labels.add(context.resources.getString(stringId))
        listeners.add(listener)
    }

    fun add(label: CharSequence, listener: OnClickListener) {
        labels.add(label)
        listeners.add(listener)
    }

    fun applyToBuilder(builder: MaterialAlertDialogBuilder) {
        builder.setItems(labels.toTypedArray()) { dialog, i -> listeners[i].onClick(dialog, i) }
    }
}
